sdk/tables/test-resources.json

```diff
@@ -15,7 +15,21 @@
           "type": "string",
           "defaultValue": "core.windows.net",
           "metadata": {
-              "description": "The url suffix to use when accessing the storage data plane."
+              "description": "The url suffix to use when accessing the storage data plane in public cloud."
+          }
+        },
+        "tablesStorageUsGovEndpointSuffix": {
+          "type": "string",
+          "defaultValue": "core.usgovcloudapi.net",
+          "metadata": {
+              "description": "The url suffix to use when accessing the storage data plane in UsGov cloud."
+          }
+        },
+        "tablesStorageChinaEndpointSuffix": {
+          "type": "string",
+          "defaultValue": "core.chinacloudapi.cn",
+          "metadata": {
+              "description": "The url suffix to use when accessing the storage data plane in China cloud."
```

Comment:
@YalinLi0312 You can actually replace all `tablesStorage*` endpoint suffix parameters (including public) with:

```
        "storageEndpointSuffix": {
          "type": "string",
          "defaultValue": "core.windows.net",
          "metadata": {
              "description": "The url suffix to use when accessing the storage data plane."
          }
        }
```

Our test resource deployment script will inject the cloud-specific value as an arm template parameter based on the cloud environment, so you don't have to do any conditionals in the template itself.

See https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/clouds/AzureUSGovernment.json#L14 for example, and we add this to the arm template parameters [here](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/New-TestResources.ps1#L678-L681).